### PR TITLE
Propagate http header callback from websocket back to client.

### PR DESCRIPTION
**Pull request by vonox7**
On Socket.IO Android we already have this callback, so backport it to iOS.

### CODE REVIEW

**Comment by vonox7** [comment]
Note: we need the current (unreleased) Starscream version, as we need the newly introduced callback from there.

**Comment by nuclearace** [comment]
@vonox7 Can you open this against development?

File: Source/SocketIO/Client/SocketIOClientSpec.swift [hunk 1]
```diff
@@ -334,4 +334,15 @@ public enum SocketClientEvent : String {
...
+    /// Emitted when when upgrading the http connection to a websocket connection.
+    ///
+    /// Usage:
+    ///
+    /// ```swift
+    /// socket.on(clientEvent: .httpResponseHeaders) {data, ack in
+    ///     // Some header logic
+    /// }
+    /// ```
+    case httpResponseHeaders
```
Comment (nuclearace) [inline Source/SocketIO/Client/SocketIOClientSpec.swift]: If this is notifying when switching transports, it should be named something `websocketUpgrade` and the payload documented.

Comment (vonox7) [reply Source/SocketIO/Client/SocketIOClientSpec.swift]: you're right, fixed in next PR

**Pull request closed without merging**